File: jzoffer 21-40/24.js
/* function TreeNode(x) {
    this.val = x;
    this.left = null;
    this.right = null;
} */
//用于记录暂时的路径
var path=[];
//用于记录完整的路径
var arr=[];
function FindPath(root, expectNumber)
{
    if(!root) return arr;
    path.push(root.val);
    if((root.val==expectNumber) && root.left==null && root.right==null){
        arr.push(path.slice());
    }
    FindPath(root.left,expectNumber-root.val);
    FindPath(root.right,expectNumber-root.val);
    if(path.length!=0){
       path.pop();
    }
    return arr;
}

File: jzoffer 21-40/24.cpp
/*
struct TreeNode {
	int val;
	struct TreeNode *left;
	struct TreeNode *right;
	TreeNode(int x) :
			val(x), left(NULL), right(NULL) {
	}
};*/
class Solution {
public:
    //用于记录路径
    vector<vector<int>> buffer;
    //记录目前的这一条路径
    vector<int> temp;
    vector<vector<int> > FindPath(TreeNode* root,int expectNumber) {
    if(root==NULL){
        return buffer;
    }
        temp.push_back(root->val);
        if((expectNumber-root->val)==0 && root->left==NULL && root->right==NULL){
            buffer.push_back(temp);
        }
        FindPath(root->left,expectNumber-root->val);
        FindPath(root->right,expectNumber-root->val);
        if(temp.size()!=0)
            temp.pop_back();
        return buffer;
    }
};

